Code/Editor/Lib/Tests/test_Main.cpp

```diff
@@ -14,24 +14,28 @@
 #include <QApplication>
 
 class EditorLibTestEnvironment
-    : public AZ::Test::ITestEnvironment
+    : public ::UnitTest::TraceBusHook
 {
 public:
     ~EditorLibTestEnvironment() override = default;
 
 protected:
     void SetupEnvironment() override
     {
+        ::UnitTest::TraceBusHook::SetupEnvironment();
+
         AZ::Environment::Create(nullptr);
         AttachEditorAZEnvironment(AZ::Environment::GetInstance());
         AZ::AllocatorInstance<AZ::SystemAllocator>::Create();
```

Comment:
nitpick: Should the initialization order be the opposite of the teardown order?

Now that SystemAllocator destruction has been moved after the detaching of the AZ::Environment, shouldn't the creation of the SystemAllocator be moved before the attaching of the AZ::Environment?

Reply:
yeah, will move the SystemAllocator creation to before the environment attachment
My concern was that the system allocator was used during detach (is not, but just in case), the same can be argued here so I will move it